avocado/core/nrunner.py

```diff
@@ -471,13 +471,63 @@ def _create_params(self):
 
         return params
 
+    @staticmethod
+    def _get_avocado_version():
+        """Return the Avocado package version, if installed"""
+        version = "unknown.unknown"
+        if PKG_RESOURCES_AVAILABLE:
+            try:
+                version = pkg_resources.get_distribution(
+                    'avocado-framework').version
+            except pkg_resources.DistributionNotFound:
+                pass
+        return version
+
+    def get_env_variables(self):
+        """Get the default AVOCADO_TEST_* environment variables
+
+        This feature was carried from the legacy runner architecture for
+        compatibility purpose.
+        """
+        outputdir = self.runnable.kwargs.get('AVOCADO_TEST_OUTPUTDIR')
+        with open("/tmp/env.txt", "w") as f:
+            f.write(str(outputdir))
+        # create work dir inside the output dir
+        workdir = os.path.join(outputdir, 'workdir')
+        os.makedirs(workdir, exist_ok=True)
+        # create fake log dir inside output dir
+        logdir = os.path.join(outputdir, 'job-results')
```

Comment:
This one makes me really uncomfortable to keep moving forward... as it's really about the job.  Maybe it's really time to break the backwards compatibility here.